cheats/rules: Refactor into partial object and set rules + Rename to newer terminology

philipaconrad opened the pull request:
## What changed?

This PR splits partial set and object rules into their own sections, and refactors the naming scheme from "partial" and "complete" rules to multi-value and single-value rules, respectively. I've included some short notes about the older terminology too.

philipaconrad committed:
cheats/rules/partial: Refactor into partial object and set rules.

This commit splits partial set and object rules into their own sections.

Signed-off-by: Philip Conrad <philip@chariot-chaser.net>

anderseknert commented on cheats/rules/partial-object/cheat.rego:
```diff
@@ -0,0 +1,10 @@
+package cheat
+
+import rego.v1
```
Why? 🙂

philipaconrad replied:
I'll freely admit that I was borrowing from the other examples on that. Will refactor shortly.

philipaconrad replied:
Fixed in the latest commits!

anderseknert reviewed (approved):
LGTM, although I don't think we should import rego.v1 in any new examples. We should also consider how we name these rule types, but leaving it at your discretion.

charlieegan3 commented on build/cheatsheet.pdf:
This generated PDF looks a bit wonky but we can come back to that once this is in.

charlieegan3 reviewed (approved):
Thanks for working on this!

philipaconrad committed:
cheats/rules: Refactor naming to single/multi-value.

This commit refactors the naming scheme for complete/partial rules to be
the more modern single/multi-value terminology.

Also included: Notes about the older terminology that still appears in
docs and learning materials.

Signed-off-by: Philip Conrad <philip@chariot-chaser.net>

philipaconrad committed:
cheats/rules: Remove `import rego.v1` from examples.

Signed-off-by: Philip Conrad <philip@chariot-chaser.net>

philipaconrad committed:
build: Regenerate the PDF.

Signed-off-by: Philip Conrad <philip@chariot-chaser.net>

philipaconrad commented:
:information_source: In keeping with the discussion we had in Slack, I've refactored how we name the rule types:

 - Complete Rules :arrow_right: Single-Value Rules
 - Partial Set Rules :arrow_right: Multi-Value Set Rules
 - Partial Object Rules :arrow_right: Multi-Value Object Rules

I've also added some short notes about "In older documentation..." describing the older naming scheme.

philipaconrad committed:
cheats/rules: Slight wording change for multi-value rules.

Signed-off-by: Philip Conrad <philip@chariot-chaser.net>

philipaconrad committed:
build: Regenerate PDF

Signed-off-by: Philip Conrad <philip@chariot-chaser.net>
